universal.c

```diff
@@ -775,12 +775,15 @@ XS(XS_PerlIO_get_layers)
         }
 
         sv = POPs;
-        gv = MAYBE_DEREF_GV(sv);
 
-        if (!gv && !SvROK(sv))
-            gv = gv_fetchsv_nomg(sv, 0, SVt_PVIO);
+        if (SvROK(sv) && SvTYPE(SvRV(sv)) == SVt_PVIO)
+            io = (IO*)SvRV(sv);
```

Comment:
`MAYBE_DEREF_GV()` does `SvGETMAGIC()` on sv, but now we're checking flags on sv without calling `SvGETMAGIC()`

Reply:
Good point, I guess I should swap the first two options around

Reply:
That should work.